include/hw/arm/guest-services/socket.h

```diff
@@ -32,6 +32,37 @@
 #include "sys/socket.h"
 #endif
 
+#ifdef __APPLE__
+#define SOCKADDR	    sockaddr
+#define SOCKADDR_IN	    sockaddr_in
```

Comment:
No need for these defines. You can use `darwin_sockaddr`/`darwin_sockaddr_in` directly. If you don't want to redefine the structs in Darwin, you can use `#define darwin_sockaddr sockaddr` and `#define darwin_sockaddr_in sockaddr_in`when `__APPLE__` is defined.